From e8a84ba9b7c7c6b11909a7f5253d572dda9e5733 Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Thu, 19 Apr 2018 11:18:18 +0300
Subject: [PATCH 3614/5242] MLK-18106: drm/mxsfb: Fix mxsfb_create_output

commit  cd2c96b5dcf8a0827b19be9d447b0a34be7b8ef1 from
https://source.codeaurora.org/external/imx/linux-imx.git

Since MXSFB driver now also supports a bridge, this object needs to be
passed to the drm_of_find_panel_or_bridge function.
This fixes a bug created during kernel 4.14 rebase process.

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/mxsfb/mxsfb_out.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/mxsfb/mxsfb_out.c b/drivers/gpu/drm/mxsfb/mxsfb_out.c
index f42e845..044d5fb 100644
--- a/drivers/gpu/drm/mxsfb/mxsfb_out.c
+++ b/drivers/gpu/drm/mxsfb/mxsfb_out.c
@@ -86,10 +86,9 @@ static void mxsfb_panel_connector_destroy(struct drm_connector *connector)
 int mxsfb_create_output(struct drm_device *drm)
 {
 	struct mxsfb_drm_private *mxsfb = drm->dev_private;
-	struct drm_panel *panel;
 	int ret;
 
-	ret = drm_of_find_panel_or_bridge(drm->dev->of_node, 0, 0, &panel, NULL);
+	ret = drm_of_find_panel_or_bridge(drm->dev->of_node, 0, 0, &mxsfb->panel, &mxsfb->bridge);
 	if (ret)
 		return ret;
 
-- 
1.7.9.5

